src/librbd/cache/pwl/AbstractWriteLog.h

```diff
@@ -141,6 +141,15 @@ class AbstractWriteLog {
   }
   void add_into_log_map(pwl::GenericWriteLogEntries &log_entries,
                         C_BlockIORequestT *req);
+  virtual void setup_resources(
+      C_WriteSameRequestT *req, uint64_t &bytes_cached, uint64_t &bytes_dirtied,
```

Comment:
Nit: I'd prefer pointers instead of references for out variables

Reply:
Done